server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -197,12 +197,11 @@ private boolean syncSegments() {
                     remoteDirectory.deleteStaleSegmentsAsync(LAST_N_METADATA_FILES_TO_KEEP);
                 }
 
-                try (GatedCloseable<SegmentInfos> segmentInfosGatedCloseable = indexShard.getSegmentInfosSnapshot()) {
+                final Tuple<GatedCloseable<SegmentInfos>, ReplicationCheckpoint> latestSegmentInfosAndCheckpoint = indexShard
+                    .getLatestSegmentInfosAndCheckpoint();
+                try (GatedCloseable<SegmentInfos> segmentInfosGatedCloseable = latestSegmentInfosAndCheckpoint.v1()) {
```

Comment:
Going to update this to first fetch the SI closeable from shard and then compute the cp instead of returning the tuple. Otherwise we run the risk of leaving the closeable open.